### PR TITLE
Update menu swipe

### PR DESCRIPTION
**Pull request by featheredtoast**
Refactor the mobile swipe events in Discourse:

Sets the styles through css properties so that behavior can be overridden in stylesheets if needed. Refactor finish animation , and implement these as simple CSS animations through a temporary animate class over the course of the animation.

Prefer element moves using translations, rather than other css properties as translations and opacity triggers are less resource intensive. From testing on my android, this makes the pull out menus MUCH smoother.

I also made all menu animations adhere to reduced motion preferences for a11y improvements.

### CODE REVIEW

**Review by pmusaraj — APPROVED** [review]
This looks great! Looking forward to trying this out on iOS. 

There is a fair bit of jQuery usage here though that we should replace by vanilla JS.

File: app/assets/javascripts/discourse/app/components/site-header.js [hunk 1]
```diff
@@ -38,17 +36,34 @@ const SiteHeaderComponent = MountWidget.extend(
     },
 
     _animateOpening($panel) {
-      $panel.css({ right: "", left: "" });
+      const $headerCloak = $(".header-cloak");
```
Comment (eviltrout) [inline app/assets/javascripts/discourse/app/components/site-header.js:39]: Maybe it's worth removing all the jQuery while you're in there?

Comment (featheredtoast) [reply app/assets/javascripts/discourse/app/components/site-header.js:39]: sure, can do!

**Review by eviltrout — APPROVED** [review]
Oooooh yeah that's nice!

**Comment by featheredtoast** [comment]
Still doing a bit more cleanup, will also attempt to look into touch tests too.

File: app/assets/javascripts/discourse/app/components/site-header.js [hunk 2]
```diff
@@ -394,27 +392,26 @@ const SiteHeaderComponent = MountWidget.extend(
...
-          if ($panelBody.prop("style").height !== "100%") {
-            $panelBody.height("100%");
+          if (panelBody.style.height !== "100%") {
+            panelBody.style.setProperty("height", "100%");
           }
-          if (style.top !== menuTop + "px" || style[heightProp] !== height) {
-            $panel.css({ top: menuTop + "px", [heightProp]: height });
-            $(".header-cloak").css({ top: menuTop + "px" });
+          if (
+            panel.style.top !== menuTop + "px" ||
```
Comment (pmusaraj) [inline app/assets/javascripts/discourse/app/components/site-header.js:399]:
```suggestion
            panel.style.top !== `${menuTop}px` ||
```

**Comment by featheredtoast** [comment]
OK, I've added tests as well as updated some code to allow those tests to run. The test environment was breaking a few things so I've cleaned some stuff up. The main points here are:

* Zoom methods on test windows -- through non-testem environments this was using a non-standard zoom property. Through testem environments, this was using a transform property. I've updated the non-testem to use more standard transforms for zooming.
* I've updated the detection for dropdown vs slide-in header modes. Mobile test detection on header should always be 'slide-in'. It takes the difference of header width vs document width, and in tests, the display window is always going to be tiny in comparison with the body.

**Comment by featheredtoast** [comment]
@eviltrout when you have a moment, would you mind doing a quick sanity check for these? Reasons for the changes is in the comment above ☝️ 

https://github.com/discourse/discourse/pull/13277/files?file-filters%5B%5D=.js#diff-4ce6a279eb43abbfe77be11767e60a0561926e29dbe440d776886b5c7f4ea01b

https://github.com/discourse/discourse/pull/13277/files#diff-c5865c5d26ff1eef7ee0e664a9eec8ca10c045c828aa90a5d47bc7def5381dadL324

**Review by eviltrout — APPROVED** [review]
Looks good to me!

**Pull request merged**